leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/RegisterResource.java

```diff
@@ -140,6 +145,26 @@ public void handleDELETE(CoapExchange exchange) {
     }
 
     private void handleRegister(CoapExchange exchange, Request request) {
+
+        // Check if this incoming request is using OSCORE
+        if (exchange.advanced().getRequest().getOptions().getOscore() != null) {
+            LOG.info("Client registered using OSCORE");
+
+            // Update the URI of the associated OSCORE Context with the client's URI
+            // So the server can send requests to the client
+            HashMapCtxDB db = HashMapCtxDB.getInstance();
+            OSCoreCtx clientCtx = db.getContext(exchange.advanced().getCryptographicContextID());
+
+            try {
+                db.addContext(request.getScheme() + "://"
+                        + request.getSourceContext().getPeerAddress().getHostString().toString(), clientCtx);
+            } catch (OSException e) {
+                LOG.error("Failed to update OSCORE Context for registering client.", request, e);
+                e.printStackTrace();
+            }
+
+        }
+
```

Comment:
This part is strange to me too but this is probably mainly because all of this is not clear to me. So we should probably discuss a bit about what could looks like the OSCORE integration in Leshan. (I will open an issue)

Reply:
let's discuss about that here https://github.com/eclipse/leshan/issues/727